### ISSUE TITLE
multipass failed creating a bridge

### ISSUE THREAD

**Issue by marosg42** [canonical/multipass#344]
sudo snap install multipass --beta --classic
There is no bridge created
syslog shows
```
Aug 20 08:02:17 krogh snapd[4144]: 2018/08/20 08:02:17.222869 api.go:1046: Installing snap "multipass" revision unset
Aug 20 08:02:19 krogh systemd[1]: Reloading.
Aug 20 08:02:19 krogh systemd[1]: Started ACPI event daemon.
Aug 20 08:02:19 krogh systemd[1]: Reloading.
Aug 20 08:02:19 krogh systemd[1]: Started ACPI event daemon.
Aug 20 08:02:19 krogh systemd[1]: Mounting Mount unit for multipass, revision 334...
Aug 20 08:02:19 krogh systemd[1]: Mounted Mount unit for multipass, revision 334.
Aug 20 08:02:20 krogh kernel: [60502.048301] audit: type=1400 audit(1534752140.342:53): apparmor="STATUS" operation="profile_replace" profile="unconfined" name="snap-update-ns.multipass" pid=588 comm="apparmor_parser"
Aug 20 08:02:20 krogh kernel: [60502.056694] audit: type=1400 audit(1534752140.354:54): apparmor="STATUS" operation="profile_replace" profile="unconfined" name="snap.multipass.hook.configure" pid=590 comm="apparmor_parser"
Aug 20 08:02:20 krogh kernel: [60502.067736] audit: type=1400 audit(1534752140.362:55): apparmor="STATUS" operation="profile_replace" profile="unconfined" name="snap.multipass.hook.install" pid=592 comm="apparmor_parser"
Aug 20 08:02:20 krogh kernel: [60502.079383] audit: type=1400 audit(1534752140.374:56): apparmor="STATUS" operation="profile_replace" profile="unconfined" name="snap.multipass.hook.post-refresh" pid=595 comm="apparmor_parser"
Aug 20 08:02:20 krogh kernel: [60502.091280] audit: type=1400 audit(1534752140.386:57): apparmor="STATUS" operation="profile_replace" profile="unconfined" name="snap.multipass.libvirt-bin" pid=598 comm="apparmor_parser"
Aug 20 08:02:20 krogh kernel: [60502.105877] audit: type=1400 audit(1534752140.402:58): apparmor="STATUS" operation="profile_replace" profile="unconfined" name="snap.multipass.multipass" pid=600 comm="apparmor_parser"
Aug 20 08:02:20 krogh kernel: [60502.115423] audit: type=1400 audit(1534752140.410:59): apparmor="STATUS" operation="profile_replace" profile="unconfined" name="snap.multipass.multipassd" pid=602 comm="apparmor_parser"
Aug 20 08:02:20 krogh kernel: [60502.127823] audit: type=1400 audit(1534752140.422:60): apparmor="STATUS" operation="profile_replace" profile="unconfined" name="snap.multipass.virsh" pid=604 comm="apparmor_parser"
Aug 20 08:02:21 krogh systemd[1]: Reloading.
Aug 20 08:02:21 krogh systemd[1]: Started ACPI event daemon.
Aug 20 08:02:21 krogh systemd[1]: Reloading.
Aug 20 08:02:21 krogh systemd[1]: Started ACPI event daemon.
Aug 20 08:02:21 krogh systemd[1]: Reloading.
Aug 20 08:02:21 krogh systemd[1]: Started ACPI event daemon.
Aug 20 08:02:22 krogh systemd[1]: Started Service for snap application multipass.libvirt-bin.
Aug 20 08:02:22 krogh systemd[1]: Started Service for snap application multipass.multipassd.
Aug 20 08:02:23 krogh systemd-udevd[833]: Could not generate persistent MAC address for mpqemubr0: No such file or directory
Aug 20 08:02:23 krogh kernel: [60505.302063] device mpqemubr0-dummy entered promiscuous mode
Aug 20 08:02:23 krogh kernel: [60505.310458] IPv6: ADDRCONF(NETDEV_UP): mpqemubr0: link is not ready
Aug 20 08:02:23 krogh dnsmasq[1056]: started, version 2.75 cachesize 150
Aug 20 08:02:23 krogh dnsmasq[1056]: compile time options: IPv6 GNU-getopt DBus i18n IDN DHCP DHCPv6 no-Lua TFTP conntrack ipset auth DNSSEC loop-detect inotify
Aug 20 08:02:23 krogh dnsmasq-dhcp[1056]: DHCP, IP range 10.11.23.2 -- 10.11.23.254, lease time infinite
Aug 20 08:02:23 krogh dnsmasq-dhcp[1056]: DHCP, sockets bound exclusively to interface mpqemubr0
Aug 20 08:02:23 krogh dnsmasq[1056]: reading /etc/resolv.conf
Aug 20 08:02:23 krogh dnsmasq[1056]: using nameserver 100.107.0.31#53
Aug 20 08:02:23 krogh dnsmasq[1056]: using nameserver 100.107.0.33#53
Aug 20 08:02:23 krogh dnsmasq[1056]: read /etc/hosts - 7 addresses
Aug 20 08:02:23 krogh dnsmasq[1056]: cannot read /var/snap/multipass/common/data/multipassd/vm-ips/dnsmasq.hosts: No such file or directory
Aug 20 08:02:24 krogh multipass.multipassd[740]: E0820 08:02:24.068918024     740 uri_parser.cc:46]           bad uri.scheme: ''
Aug 20 08:02:24 krogh multipass.multipassd[740]: E0820 08:02:24.069003804     740 uri_parser.cc:52]                            ^ here
Aug 20 08:02:24 krogh multipass.multipassd[740]: E0820 08:02:24.069012234     740 http_proxy.cc:54]           cannot parse value of 'http_proxy' env var
Aug 20 08:02:24 krogh multipassd[740]: gRPC listening on unix:/run/multipass_socket
Aug 20 08:02:24 krogh multipassd[740]: QIODevice::write (QFile, "/var/snap/multipass/common/cache/multipassd/vault/multipassd-image-records.json"): device not open
Aug 20 08:02:24 krogh multipassd[740]: QSslSocket: cannot resolve OPENSSL_init_ssl
Aug 20 08:02:24 krogh multipassd[740]: QSslSocket: cannot resolve OPENSSL_init_crypto
Aug 20 08:02:24 krogh multipassd[740]: QSslSocket: cannot resolve ASN1_STRING_get0_data
Aug 20 08:02:24 krogh multipassd[740]: QSslSocket: cannot resolve EVP_CIPHER_CTX_reset
Aug 20 08:02:24 krogh multipassd[740]: QSslSocket: cannot resolve RSA_bits
Aug 20 08:02:24 krogh multipassd[740]: QSslSocket: cannot resolve OPENSSL_sk_new_null
Aug 20 08:02:24 krogh multipassd[740]: QSslSocket: cannot resolve OPENSSL_sk_push
Aug 20 08:02:24 krogh multipassd[740]: QSslSocket: cannot resolve OPENSSL_sk_free
Aug 20 08:02:24 krogh multipassd[740]: QSslSocket: cannot resolve OPENSSL_sk_num
Aug 20 08:02:24 krogh multipassd[740]: QSslSocket: cannot resolve OPENSSL_sk_pop_free
Aug 20 08:02:24 krogh multipassd[740]: QSslSocket: cannot resolve OPENSSL_sk_value
Aug 20 08:02:24 krogh multipassd[740]: QSslSocket: cannot resolve DH_get0_pqg
Aug 20 08:02:24 krogh multipassd[740]: QSslSocket: cannot resolve SSL_CTX_set_options
Aug 20 08:02:24 krogh multipassd[740]: QSslSocket: cannot resolve SSL_get_client_random
Aug 20 08:02:24 krogh multipassd[740]: QSslSocket: cannot resolve SSL_SESSION_get_master_key
Aug 20 08:02:24 krogh multipassd[740]: QSslSocket: cannot resolve SSL_session_reused
Aug 20 08:02:24 krogh multipassd[740]: QSslSocket: cannot resolve TLS_method
Aug 20 08:02:24 krogh multipassd[740]: QSslSocket: cannot resolve TLS_client_method
Aug 20 08:02:24 krogh multipassd[740]: QSslSocket: cannot resolve TLS_server_method
Aug 20 08:02:24 krogh multipassd[740]: QSslSocket: cannot resolve X509_STORE_CTX_get0_chain
Aug 20 08:02:24 krogh multipassd[740]: QSslSocket: cannot resolve X509_getm_notBefore
Aug 20 08:02:24 krogh multipassd[740]: QSslSocket: cannot resolve X509_getm_notAfter
Aug 20 08:02:24 krogh multipassd[740]: QSslSocket: cannot resolve X509_get_version
Aug 20 08:02:24 krogh multipassd[740]: QSslSocket: cannot resolve OpenSSL_version_num
Aug 20 08:02:24 krogh multipassd[740]: QSslSocket: cannot resolve OpenSSL_version
Aug 20 08:02:24 krogh multipassd[740]: Incompatible version of OpenSSL
```
It looks like there are issues with detecting proxy

ubuntu@krogh:~$ env|grep proxy
http_proxy=http://100.107.0.4:1080/
https_proxy=http://100.107.0.4:1080/
no_proxy=100.107.0.207,100.107.0.31,100.107.0.41,100.68.1.4,100.86.1.4,127.0.0.1,localhost

ubuntu@krogh:~$ cat /etc/environment 
PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games"
http_proxy="http://100.107.0.4:1080"
https_proxy="http://100.107.0.4:1080"

**Comment by Saviq** [comment]
@marosg42 as far as proxy goes, we support `proxy` as a snap setting:

`snap set multipass proxy=http://100.107.0.4:1080`

I believe the values from `/etc/environment` don't propagate to snap services, can you check like below?

`sudo cat /proc/`pidof multipassd`/environ | tr '\0' '\n' | grep no_proxy`

All that said, I don't think that the lack of proxy would result in the other issues you're seeing.

Can you verify the version of multipass installed?

`multipass version`

**Comment by marosg42** [comment]
I just checked with --edge, it worked, mpqemubr0 was created

**Comment by marosg42** [comment]
regarding proxy settings...

for snapd to be able to install snaps, I had to modify /etc/environment and restart snapd, that works

then I didsnap install, which failed to create bridge as shown above

when I do multipass launch it reports error becasue it cannot get to images, for that I used snap set multipass proxy=... and then it can download

after that it reports it cannot connect to instance, which of course it cannot as bridge was not created

I have no idea why there is any check for proxy in the code while bridge is created

and as I said, edge works

**Comment by Saviq** [comment]
Can you try with the `libvirt` backend?

`snap set mulitpass driver=LIBVIRT`

**Comment by Saviq** [comment]
This will get solved with #626 and #611.
